pkg/app/server/grpcapi/api.go

```diff
@@ -787,23 +788,26 @@ func (a *API) GetPlanPreviewResults(ctx context.Context, req *apiservice.GetPlan
 			return nil, status.Error(codes.FailedPrecondition, fmt.Sprintf("Command %s is not a plan preview command", commandID))
 		}
 
+		if _, isThere := pipedNameMap[cmd.PipedId]; !isThere {
```

Comment:
Sorry, I could not get why this needs to use map.

Reply:
The function loops about commandID twice in lines 774 and 839.
I thought it would be useless to call getPiped for the same piped.id in each loop, so I used map to make a memoization for the first getPiped.

Reply:
Ah, this if statement is to save calling getPiped if you have the same piped for different commandIDs.

Reply:
I got your point, thanks.